### PR TITLE
Treasure Knight

### PR DESCRIPTION
**Pull request by Mach131**

### CODE REVIEW

File: scenes/game/local_game.gd [hunk 1]
```diff
@@ -5994,6 +6083,31 @@ func handle_strike_effect(card_id : int, effect, performing_player : Player):
...
+			var starting_range = abs(previous_closest_location - origin)
+			if performing_player.is_left_of_location(previous_closest_location):
+				if starting_range >= target_range:
+					# Past target range; should end on the right
+					target_closest_location = min(origin + target_range, MaxArenaLocation)
+				else:
+					# Closer than target range; should end on left
+					target_closest_location = max(origin - target_range, MinArenaLocation)
+			else:
+				# vice-versa
```
Comment (daniel-k-taylor) [inline scenes/game/local_game.gd:6101]: just write the actual thing in the comment: "If player is to the right"

File: scenes/game/local_game.gd [hunk 2]
```diff
@@ -5994,6 +6083,31 @@ func handle_strike_effect(card_id : int, effect, performing_player : Player):
 			if not nowhere_to_pull:
 				change_game_state(Enums.GameState.GameState_PlayerDecision)
 				events += [create_event(Enums.EventType.EventType_ChooseArenaLocationForEffect, performing_player.my_id, 0)]
+		"pull_to_range":
```
Comment (daniel-k-taylor) [inline scenes/game/local_game.gd:6086]: make sure you have a test that is like, treasure at location 3, wide tinker at 4/5/6, and you're trying to do this effect.

**Review by daniel-k-taylor — CHANGES_REQUESTED** [review]

File: data/card_definitions.json [hunk 3]
```diff
@@ -22932,6 +22932,388 @@
...
+			"display_name": "Secure Vault",
+			"effects": [
+				{
+					"timing": "hit",
+					"effect_type": "force_for_effect",
+					"per_force_effect": {
+						"effect_type": "add_top_discard_to_gauge",
+						"amount": 1,
+						"and": {
+							"effect_type": "draw_for_card_in_gauge",
```
[the remark this answers is not in the document]
Comment (Mach131) [reply data/card_definitions.json:23014]: Yup, there was a brief discussion in beginner-lounge

File: scenes/game/local_game.gd [hunk 4]
```diff
@@ -1702,8 +1738,11 @@ class Player:
 			events += remove_buddy(buddy_id)
 		return events
 
-	func get_force_with_cards(card_ids : Array, reason : String, treat_ultras_as_single_force : bool):
-		var force_generated = free_force
+	func get_force_with_cards(card_ids : Array, reason : String, treat_ultras_as_single_force : bool, use_free_force : bool):
```
[the remark this answers is not in the document]
Comment (Mach131) [reply scenes/game/local_game.gd:1741]: I actually started by just renaming the old free_force variable and making sure nothing changed about Hazama tests before adding anything new; is there anything in particular you think may not be covered?

File: scenes/game/local_game.gd [hunk 5]
```diff
@@ -4028,7 +4083,7 @@ func handle_strike_effect(card_id : int, effect, performing_player : Player):
 			var actual_amount = min(amount, len(performing_player.discards))
 			if actual_amount > 0:
 				var card_ids = []
-				for i in range(performing_player.discards.size() - 1, -1, -1):
```
[the remark this answers is not in the document]
Comment (Mach131) [reply scenes/game/local_game.gd:4031]: yeah whoops

File: scenes/game/local_game.gd (hunk 2 above)
Comment (Mach131) [reply scenes/game/local_game.gd:6086]: have a set of tinker tank tests yup, though I should probably also check when treasure's at location 4

File: scenes/game/local_game.gd (hunk 4 above)
[the remark this answers is not in the document]
Comment (daniel-k-taylor) [reply scenes/game/local_game.gd:1744]: Nothing in particular. Just make sure it all passes and also do a 100 random hazama run.

**Review by daniel-k-taylor — APPROVED** [review]

File: scenes/game/local_game.gd (hunk 4 above)
[the remark this answers is not in the document]
Comment (Mach131) [reply scenes/game/local_game.gd:1744]: yup, all passing

**Pull request merged**